handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -148,9 +154,45 @@ public PcapWriteHandler(OutputStream outputStream) {
      *                              {@link NullPointerException}
      */
     public PcapWriteHandler(OutputStream outputStream, boolean captureZeroByte, boolean writePcapGlobalHeader) {
+        this(outputStream, captureZeroByte, writePcapGlobalHeader, false);
+    }
+
+    /**
+     * Create new {@link PcapWriteHandler} Instance
+     *
+     * @param outputStream          OutputStream where Pcap data will be written. Call {@link #close()} to close this
+     *                              OutputStream.
+     * @param captureZeroByte       Set to {@code true} to enable capturing packets with empty (0 bytes) payload.
+     *                              Otherwise, if set to {@code false}, empty packets will be filtered out.
+     * @param writePcapGlobalHeader Set to {@code true} to write Pcap Global Header on initialization.
+     *                              Otherwise, if set to {@code false}, Pcap Global Header will not be written
+     *                              on initialization. This could when writing Pcap data on a existing file where
+     *                              Pcap Global Header is already present.
+     * @param sharedOutputStream    Set to {@code true} if multiple {@link PcapWriteHandler} instances will be
+     *                              writing to the same {@link OutputStream} concurrently, and write locking is
+     *                              required. Otherwise, if set to {@code false}, no locking will be done.
+     *                              Additionally, {@link #close} will not close the underlying {@code OutputStream}.
+     *                              Note: it is probably an error to have both {@code writePcapGlobalHeader} and
+     *                              {@code sharedOutputStream} set to {@code true} at the same time.
+     * @throws NullPointerException If {@link OutputStream} is {@code null} then we'll throw an
+     *                              {@link NullPointerException}
+     */
+    public PcapWriteHandler(OutputStream outputStream, boolean captureZeroByte, boolean writePcapGlobalHeader,
+            boolean sharedOutputStream) {
         this.outputStream = ObjectUtil.checkNotNull(outputStream, "OutputStream");
         this.captureZeroByte = captureZeroByte;
         this.writePcapGlobalHeader = writePcapGlobalHeader;
+        this.sharedOutputStream = sharedOutputStream;
+    }
+
+    /**
+     * Writes the Pcap Global Header to the provided {@code OutputStream}
+     *
+     * @param outputStream OutputStream where Pcap data will be written.
+     * @throws IOException if there is an error writing to the {@code OutputStream}
+     */
+    public static void writeGlobalHeader(OutputStream outputStream) throws IOException {
+        outputStream.write(PcapHeaders.GLOBAL_HEADER);
     }
```

Comment:
We don't need this anymore because Pcap Global Header is already been written in `PcapWriter`.

Reply:
Well, not if all the PcapWriters are constructed with the flag to NOT write the header. i.e assuming that the header has already been written, which is already an option. The problem being, who then writes that header? It could be that one PcapWriter is constructed with the flag to write the header, but then you need to be sure that this PcapWriter is the first one to actually be activated!
This method is provided to allow the header to be written in advance of any PcapWriters being constructed.